src/Microsoft.SqlTools.ServiceLayer/NotebookConvert/NotebookConvertService.cs

```diff
@@ -200,8 +201,13 @@ internal static NotebookDocument ConvertSqlToNotebook(string sql)
                         commentBlock = commentBlock.Remove(commentBlock.Length - 2);
                     }
 
-                    doc.Cells.Add(GenerateMarkdownCell(commentBlock.Trim()));
-                } else if (fragment.TokenType == NotebookTokenType.SinglelineComment)
+                    // Trim off extra spaces for each line. This helps keep comment asterisks aligned on the 
```

Comment:
Do you know why the newlines are inserted?

Reply:
For which part?

Reply:
For which part?

Reply:
The issue you called out - where newlines are added when you convert sql -> notebook -> back to sql

Reply:
Not sure. It only affects multi-line markdown cells, but we already have newlines included for those in notebooks. We weren't inserting extra newlines for code cells. Seems like the sort of thing that was added as an afterthought. Usually you would need to insert your own newlines when combining an array of text.

Reply:
Looks like you were the one who added the notebook conversion code a year ago.

Reply:
Nope, this was all pretty quickly done hackathon-type stuff so I'm not surprised there's some issues like that. Not a huge deal, was just curious if this maybe pointed to a bigger issue.